src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverService.findByLogin(login);
+        if (driver.isPresent() && driver.get().getPassword().equals(password)) {
+            return driver.get();
+        }
+        throw new AuthenticationException("Password is incorrect!");

```

Comment:
if login was incorrect we will receive the same message, let's change it to "Login or password is incorrect"